backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -233,18 +256,22 @@ public boolean updateEndpoint(Endpoint endpoint) {
                 case WEBHOOK:
                     WebhookProperties properties = endpoint.getProperties(WebhookProperties.class);
                     return entityManager.createQuery(webhookQuery)
+                            .setParameter("basicAuthenticationSourcesId", properties.getBasicAuthenticationSourcesId())
```

Comment:
I don't think we should migrate the credentials to Sources by modifying the public REST API, especially when the changes here will never be used with a public REST request (the sources IDs are marked as `@JsonIgnore`).

I'd prefer a temporary REST internal endpoint dedicated to the migration. It would rely on DB queries very similar to what you already wrote, but with a cleaner separation between public and internal matters.

Reply:
Here are two examples of internal endpoints dedicated to migrations:
- https://github.com/RedHatInsights/notifications-backend/blob/master/engine/src/main/java/com/redhat/cloud/notifications/templates/EmailTemplateMigrationService.java
- https://github.com/RedHatInsights/notifications-backend/blob/master/backend/src/main/java/com/redhat/cloud/notifications/routers/internal/SubscriptionToEventTypeMigrationService.java

Reply:
All right! I have created everything ad-hoc so that we can simply wipe it out easily once we're done with the migrations, as you suggested.